experiment/simulation/js/main.js

```diff
@@ -1 +1,490 @@
+    xygrid.pop();
+  }
+});
+
+xz_grid.addEventListener("click", () => {
+  if (xz_grid.checked) {
+    let grid = new THREE.GridHelper(size, divisions);
+    grid.geometry.rotateZ(Math.PI / 2);
+    xzgrid.push(grid);
+    scene.add(xzgrid[0]);
+  } else {
+    scene.remove(xzgrid[0]);
+    xzgrid.pop();
+  }
+});
+
+yz_grid.addEventListener("click", () => {
+  if (yz_grid.checked) {
+    let grid = new THREE.GridHelper(size, divisions);
+    let vector3 = new THREE.Vector3(0, 1, 0);
+    grid.lookAt(vector3);
+    yzgrid.push(grid);
+    scene.add(yzgrid[0]);
+  } else {
+    scene.remove(yzgrid[0]);
+    yzgrid.pop();
+  }
+});
+
+let buttons = document.getElementsByTagName("button");
+const size = 50;
+const divisions = 25;
+
+let span_add_modal = document.getElementsByClassName("close")[1];
+
+span_add_modal.onclick = function () {
+  modal_add.style.display = "none";
+};
+
+document.getElementById("add-shape-btn").onclick = function () {
+  modal_add.style.display = "block";
+
+  document.querySelector(".buttonissecondary").addEventListener("click", () => {
+    let xcoord = document.getElementById("x1").value;
+    let ycoord = document.getElementById("y1").value;
+    let zcoord = document.getElementById("z1").value;
+    no_of_shapes++;
+    console.log(no_of_shapes);
+    is_shapeexist = 1;
+    console.log(document.getElementById("shape-add-dropdown").value);
+    if (document.getElementById("shape-add-dropdown").value == "Cube") {
+      createCube(
+        xcoord,
+        ycoord,
+        zcoord,
+        shapes,
+        scene,
+        point,
+        shapevertex,
+        dragx,
+        dragy,
+        dragz
+      );
+    }
+    if (document.getElementById("shape-add-dropdown").value == "Tetrahedron") {
+      createTetrahedron(
+        xcoord,
+        ycoord,
+        zcoord,
+        shapes,
+        scene,
+        point,
+        shapevertex,
+        dragx,
+        dragy,
+        dragz
+      );
+    }
+    if (document.getElementById("shape-add-dropdown").value == "Octahedron") {
+      createOctahedron(
+        xcoord,
+        ycoord,
+        zcoord,
+        shapes,
+        scene,
+        point,
+        shapevertex,
+        dragx,
+        dragy,
+        dragz
+      );
+    }
+    if (document.getElementById("shape-add-dropdown").value == "Dodecahedron") {
+      createDodecahedron(
+        xcoord,
+        ycoord,
+        zcoord,
+        shapes,
+        scene,
+        point,
+        shapevertex,
+        dragx,
+        dragy,
+        dragz
+      );
+    }
+    modal_add.style.display = "none";
+  });
+};
+
+let raycaster = new THREE.Raycaster();
+let mouse = new THREE.Vector2();
+let plane = new THREE.Plane();
+let pNormal = new THREE.Vector3(0, 1, 0);
+
+let planeIntersect = new THREE.Vector3();
+let pIntersect = new THREE.Vector3();
+let shift = new THREE.Vector3();
+let isDragging = false;
+let dragObject;
+let point = [];
+let shapevertex = [];
+let dot_list = [];
```

Comment:
Why do we need this list when we always have just 1 element in it?

Reply:
If we don't add the object to dot_list, we are facing an error.